### PR TITLE
Feature/Coupon CRUD

### PR DESCRIPTION
**Pull request by namirinz**

### CODE REVIEW

**Review by Npwskp — DISMISSED** [review]
LGTM !

File: src/schema/coupons.schema.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  @ApiProperty({ type: String, description: 'Coupon Description' })
+  description: string;
+
+  @Prop({ default: null })
+  @ApiProperty({
+    type: String,
+    isArray: true,
+    description: 'Coupon Required Menus',
+  })
+  required_menus: Types.ObjectId[];
```
Comment (ishiasaka) [inline src/schema/coupons.schema.ts]: Please reference to menu

File: src/schema/coupons.schema.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  amount: number;
+
+  @Prop({ default: false })
+  @ApiProperty({
+    type: Boolean,
+    description: 'Coupon status',
+  })
+  activated: boolean;
+
+  @Prop({ default: null })
```
Comment (ishiasaka) [inline src/schema/coupons.schema.ts]: default = 0

**Review by ishiasaka — CHANGES_REQUESTED** [review]

File: src/schema/coupons.schema.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  @Prop({ required: true })
+  @ApiProperty({ type: String, description: 'Coupon Code' })
+  title: string;
+
+  @Prop({ default: null })
+  @ApiProperty({ type: String, description: 'Coupon Description' })
+  description: string;
+
+  @Prop({ default: null, ref: () => MenuSchema })
```
Comment (ishiasaka) [inline src/schema/coupons.schema.ts]: default: `[]`

**Review by ishiasaka — CHANGES_REQUESTED** [review]

**Review by ishiasaka — APPROVED** [review]
すごいです

**Pull request merged**